packages/in_app_purchase/in_app_purchase_storekit/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -26,7 +26,7 @@ @interface InAppPurchasePlugin ()
 // Callback channel to dart used for when a function from the payment queue delegate is triggered.
 @property(strong, nonatomic, readonly) FlutterMethodChannel *paymentQueueDelegateCallbackChannel;
 
-@property(strong, nonatomic, readonly) NSObject<FlutterTextureRegistry> *registry;
+@property(weak, nonatomic, readonly) NSObject<FlutterTextureRegistry> *registry;
```

Comment:
Looks like registry is unused in the plugin code and can be completely removed. "messenger" can also be removed with a little tweak in the code.

And it looks like registrar can be weak as the `engine` should keep a strong reference to the registrar and plugin should out live the `engine`.